# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1208291409 14400
#      Tue Apr 15 16:30:09 2008 -0400
# Node ID a626db2e8a1c5fd965d49c8e91fc101cdcc33146
# Parent  263bcc319233d269d851fc48a39243dc832d7419
view: get values from current axes if nargin == 0

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-04-15  John W. Eaton  <jwe@octave.org>
+
+	* plot/view.m: Get values from current axes if nargin == 0.
+
 2008-04-015  David Bateman  <dbateman@free.fr>
 
 	* plot/__patch__.m: Fix for NaN values in faces of patches.
 	
 2008-04-14  David Bateman  <dbateman@free.fr>
 
 	* plot/rose.m: Add missing comment mark that causes issues with
 	octave-forge function referencing.
diff --git a/scripts/plot/view.m b/scripts/plot/view.m
--- a/scripts/plot/view.m
+++ b/scripts/plot/view.m
@@ -23,17 +23,21 @@
 ## Set or get the viewpoint for the current axes.
 ## @end deftypefn
 
 ## Author: jwe
 
 function [azimuth, elevation] = view (x, y, z)
 
   if (nargin < 4)
-    if (nargin == 1)
+    if (nargin == 0)
+      tmp = get (gca (), "view");
+      az = tmp(1);
+      el = tmp(2);
+    elseif (nargin == 1)
       if (x == 2)
 	az = 0;
 	el = 90;
       elseif (x == 3)
 	az = -37.5;
 	el = 30;
       else
 	error ("view: expecting single argument to be 2 or 3");
